Update 02-branching.md (#57)
MetOffice/git-working-practices: episodes/02-branching.md

Once the feature is done, they submit a pull request and
merge it into the `main` branch after review.
Feature branches should be relatively short-lived.

> Each collaborator must be given **write** level, or higher, permissions

Reviewer: ```suggestion
Each collaborator must have **write** level, or higher, permissions
```

on the repository.
There are 2 repositories for each person to be mindful of, ***local*** and ***origin***.

#### Pros

> - Each feature is developed away from `main` so you don't affect production code.
> - Multiple features can be developed in parallel feature branches.

Reviewer: ```suggestion
- Developers create each feature  away from `main` so you don't affect production code.
- Developers can create many features can in parallel feature branches.
```

- It's a simple model that's easy for those new to git and your project.
- Easy to set up with continuous integration testing and deployment.

#### Cons

- If you don't regularly merge changes to `main` into your feature branch
  it can become outdated, leading to merge conflicts.
- You may struggle if you need to maintain multiple production versions
  simultaneously in the same repository.

The Feature Branch model is sometimes called GitHub Flow.

> Note that all branches and commits exist within the single GitHub, ***origin***, repository
> (and are therefore easily visible by anyone who has read access to the repository)
> and any contributor can write to any branch that has not been explicitly protected.

Reviewer: ```suggestion
Note that all branches and commits exist within the single GitHub, origin repository.
They are  visible to anyone who has read access to the repository.
Any contributor can write to any branch that is not explicitly protected.
```

#### Pros

- There is a clear purpose for each branch.
- Handles complex projects well.

#### Cons

> - Very steep learning curve, not suitable for novices.

Reviewer: I don't think that we want to word this so strongly.

```suggestion
- Steeper learning curve, novices may require more help.
```
